### PR TITLE
Use public implementation MarshalSupport

### PR DESCRIPTION
**Pull request by luqunl**
Use public implementation MarshalSupport exposed in S.P.corelib.

Coreclr PR:https://github.com/dotnet/coreclr/pull/18530

### CODE REVIEW

File: src/System.Runtime.WindowsRuntime/src/System/Runtime/InteropServices/WindowsRuntime/MarshalingHelpers.cs [hunk 1]
```diff
@@ -162,7 +240,7 @@ internal static PropertyChangedEventArgs ConvertToManaged(IntPtr nativeArgsIP)
             if (nativeArgsIP == IntPtr.Zero)
                 return null;
 
-            object obj = System.StubHelpers.InterfaceMarshaler.ConvertToManagedWithoutUnboxing(nativeArgsIP);
+            object obj = InterfaceMarshalerSupport.ConvertToManagedWithoutUnboxing(nativeArgsIP);
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System/Runtime/InteropServices/WindowsRuntime/MarshalingHelpers.cs]: Should this be ` Marshal.GetUniqueObjectForIUnknown` as well?

**Comment by jkotas** [comment]
```
error BCL0015: combase.dll!RoGetActivationFactory is not supported on one\more targeted platforms.Consider using api-ms-win-core-winrt-l1-1-0.dll!RoGetActivationFactory instead
```

**Comment by luqunl** [comment]
Update PinvokeAnalyzer_Win32Apis.txt to allow  api-ms-win-core-winrt-l1-1-0.dll!RoGetActivationFactory for netcoreapp

File: src/Common/src/Interop/Windows/Interop.Libraries.cs [hunk 2]
```diff
@@ -32,5 +32,6 @@ internal static partial class Libraries
         internal const string Ws2_32 = "ws2_32.dll";
         internal const string Wtsapi32 = "wtsapi32.dll";
         internal const string CompressionNative = "clrcompression.dll";
+        internal const string CORE_WINRT = "api-ms-win-core-winrt-l1-1-0.dll";
```
Comment (jkotas) [inline src/Common/src/Interop/Windows/Interop.Libraries.cs]: This casing is different from everything else in this file. What about `CoreWinRT` ?

File: src/System.Runtime.WindowsRuntime/src/System/Runtime/InteropServices/WindowsRuntime/MarshalingHelpers.cs [hunk 3]
```diff
@@ -89,12 +111,43 @@ internal static IntPtr ConvertToNative(NotifyCollectionChangedEventArgs managedA
...
-            return System.StubHelpers.EventArgsMarshaler.CreateNativeNCCEventArgsInstance(
-                        (int)managedArgs.Action,
-                        managedArgs.NewItems,
-                        managedArgs.OldItems,
-                        managedArgs.NewStartingIndex,
-                        managedArgs.OldStartingIndex);
+            IntPtr newItemsIP = IntPtr.Zero;
+            IntPtr oldItemsIP = IntPtr.Zero;
+
+            RuntimeHelpers.PrepareConstrainedRegions();
```
Comment (jkotas) [inline src/System.Runtime.WindowsRuntime/src/System/Runtime/InteropServices/WindowsRuntime/MarshalingHelpers.cs]: Nit: `PrepareConstrainedRegions` does nothing in CoreCLR. We do not bother with these annotations anymore.

**Comment by jkotas** [comment]
@luqun Could you please resolve the conflicts and address the feedback so that this can be merged? This is needed for the new CoreCLR to get in without workarounds.

**Comment by luqunl** [comment]
> Could you please resolve the conflicts and address the feedback so that this can be merged?
@jkotas, resolved and updated. BTW, I also add an analyzerdata file, since the https://github.com/dotnet/buildtools/pull/2078 is waiting for review.

File: src/System.Runtime.WindowsRuntime/src/System/Runtime/InteropServices/WindowsRuntime/MarshalingHelpers.cs (hunk 3 above)
Comment (luqunl) [reply src/System.Runtime.WindowsRuntime/src/System/Runtime/InteropServices/WindowsRuntime/MarshalingHelpers.cs]: Removed

File: src/Common/src/Interop/Windows/Interop.Libraries.cs (hunk 2 above)
Comment (luqunl) [reply src/Common/src/Interop/Windows/Interop.Libraries.cs]: Renamed. The CORE_WINRT was copied from CoreRT repro..

**Pull request merged**